go-controller/pkg/node/gateway_init_linux_test.go

```diff
@@ -1202,16 +1203,16 @@ var _ = Describe("Gateway unit tests", func() {
 		})
 
 		It("Fails if node IP is not in host subnets", func() {
-			_, dpuSubnet, _ := net.ParseCIDR("10.0.0.101/24")
-			nodeIP := net.ParseIP("10.0.1.11")
+			_, dpuSubnet, _ := utilnet.ParseCIDRSloppy("10.0.0.101/24")
+			nodeIP := utilnet.ParseIPSloppy("10.0.1.11")
```

Comment:
You don't need to use the sloppy functions for const strings that you know are valid.

`net.ParseIP` and `net.ParseCIDR` should still be the default for parsing IPs/CIDRs. The sloppy versions should _only_ be used in contexts where you need to accept invalid strings as user input for backward-compatibility reasons. (Basically, the parameter passed to `ParseIPSloppy` or `ParseCIDRSloppy` should almost always be exactly one of the items in your list in the initial PR description (`service.Spec.ClusterIP`, `endpointSlice.Endpoints[*].Addresses`, etc)